UI.AutomationTests/EmailNotifications/EmailNotificationService.cs

```diff
@@ -58,25 +60,41 @@ public class EmailNotificationService
             "3210895a-c096-4029-b43e-9fde4642a254"
         }
     };
-    
+
     public async Task<string> GetTempPasswordForUser(string contactEmail)
     {
         var allNotifications = await NotifyApiClient.GetNotificationsAsync("email");
         var newUserEmails = allNotifications.notifications.Find(x =>
             x.emailAddress == contactEmail && 
-            x.template.id == _emails[EmailTemplates.SecondEmailNewUserConfirmation]);
+            x.template.id == _emails[EmailTemplates.SecondEmailNewUserConfirmation] &&
+            DateTime.Parse(x.createdAt, DateTimeFormatInfo.CurrentInfo) > DateTime.UtcNow.AddMinutes(-2));
         return _tempPassword.Match(newUserEmails!.body).Value;
     }
-    
+
+    public async Task PullNotificationList()
+    {
+        _notificationList = await NotifyApiClient.GetNotificationsAsync("email");
+    }
+
     public async Task ValidateEmailReceived(string contactEmail, EmailTemplates emailTemplate)
```

Comment:
Using polly to retry would be nicer here. I think in other places where we Thread.Sleep can be replaced with an API assertion before we make a UI check too

```suggestion
    private async Task<bool> QueryNotifyForEmail(string contactEmail, EmailTemplates emailTemplate)
{
    var retryPolicy = Policy
        .HandleResult<bool>(result => !result)
        .WaitAndRetryAsync(3, retryAttempt => TimeSpan.FromSeconds(10));

    return await retryPolicy.ExecuteAsync(async () =>
    {
        await PullNotificationList();
        return _notificationList.notifications.Exists(x => x.emailAddress == contactEmail &&
                                                           x.template.id == _emails[emailTemplate] &&
                                                           DateTime.Parse(x.sentAt, DateTimeFormatInfo.CurrentInfo) > DateTime.UtcNow.AddMinutes(-5));
    });
}
```